File: workspace/src/my_py_pkg/my_py_pkg/acc_publisher.py
import rclpy
from rclpy.executors import ExternalShutdownException
from rclpy.node import Node
from my_interfaces.msg import Accelerometer

class AccPublisher(Node):
    """ Publisher example that periodically sends out dummy accelerometer data """

    def __init__(self):
        super().__init__("minimal_publisher")

        self._publisher = self.create_publisher(Accelerometer, "my_acc", 10)
        self._timer = self.create_timer(0.5, self._timer_callback)

        self._counter = 0

    def _timer_callback(self):
        """ Publishes a simple message to topic """

        msg = Accelerometer()
        msg.x = 0.5
        msg.y = 0.1
        msg.z = -9.81

        self._publisher.publish(msg)
        self.get_logger().info(f"Publishing: ({msg.x} {msg.y} {msg.z})")

        self._counter += 1

def main(args = None):
    """ Main entrypoint """

    node = None
    try:
        rclpy.init()
        node = AccPublisher()
        rclpy.spin(node)
    except (KeyboardInterrupt, ExternalShutdownException):
        pass
    finally:
        if node is not None:
            node.destroy_node()
        if rclpy.ok():
            rclpy.shutdown()

if __name__ == "__main__":
    main()

File: workspace/src/my_cpp_pkg/src/acc_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "my_interfaces/msg/accelerometer.hpp"

class AccPublisher : public rclcpp::Node
{

public:
  
  AccPublisher() : Node{"acc_publisher"}, counter_{0} // Call the parent constructor from the initilizer list 
  {

    publisher_ = this->create_publisher<my_interfaces::msg::Accelerometer>("my_acc", 10);

    timer_ = this->create_wall_timer(std::chrono::milliseconds(500), std::bind(&AccPublisher::timer_callback, this));

  }

private:
  // Member varibles
  rclcpp::Publisher<my_interfaces::msg::Accelerometer>::SharedPtr publisher_;
  rclcpp::TimerBase::SharedPtr timer_;
  size_t counter_;

  // Private methods
  void timer_callback() {
    
    auto msg = my_interfaces::msg::Accelerometer();
    msg.x = 0.5;
    msg.y = 0.1;
    msg.z = -9.81;

    publisher_->publish(msg);
    RCLCPP_INFO(this->get_logger(), "Publishing: x=%.2f, y=%.2f, z=%.2f", msg.x, msg.y, msg.z);

    counter_++;
  }
};

// Entry point
int main(int argc, char* argv[]) {
  
  rclcpp::init(argc, argv);

  auto node = std::make_shared<AccPublisher>();
  rclcpp::spin(node);

  rclcpp::shutdown();

  return 0;
}